optaplanner-quickstarts/spring-boot-school-timetabling/src/main/java/com/example/persistence/ProblemChangedRepositoryEventListener.java

```diff
@@ -0,0 +1,56 @@
+package com.example.persistence;
+
+import org.optaplanner.core.api.solver.SolverStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.core.annotation.HandleBeforeCreate;
+import org.springframework.data.rest.core.annotation.HandleBeforeDelete;
+import org.springframework.data.rest.core.annotation.HandleBeforeSave;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+import org.springframework.stereotype.Component;
+
+import com.example.domain.Lesson;
+import com.example.domain.Room;
+import com.example.domain.Timeslot;
+import com.example.solver.TimeTableController;
+
+@Component
+@RepositoryEventHandler
+public class ProblemChangedRepositoryEventListener {
```

Comment:
I think it's worth describing what the issue is that a user needs to provide such a handler. This class is some kind of workaround. If I am required to put such a workaround in my project, I need to know why.

Reply:
Agreed, this needs a motivation